src/types/redis_geo.cc

```diff
@@ -85,11 +85,18 @@ rocksdb::Status Geo::Radius(const Slice &user_key, double longitude, double lati
   rocksdb::Status s = ZSet::GetMetadata(ns_key, &metadata);
   if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
 
+  GeoShape geo_shape;
+  geo_shape.type = CIRCULAR;
+  geo_shape.xy[0] = longitude;
+  geo_shape.xy[1] = latitude;
+  geo_shape.radius = radius_meters;
+  geo_shape.conversion = 1;
+
   /* Get all neighbor geohash boxes for our radius search */
-  GeoHashRadius georadius = GeoHashHelper::GetAreasByRadiusWGS84(longitude, latitude, radius_meters);
+  GeoHashRadius georadius = GeoHashHelper::GetAreasByShapeWGS84(&geo_shape);
```

Comment:
@infdahai , @git-hulk I wanted an opinion on this.

Should I change the existing georadius functions to use the new shape specific functions or should I let them be for now and they will be taken as part of a different refactor?

If taken as different refactor, I'll revert this piece of code to use `GetAreasByRadiusWGS84`.

Reply:
Ps: This new implementation is different, hence breaking some tests.

Reply:
I think it is ok to refactor them if it is convenient.